### PR TITLE
Dev/inference

### PR DESCRIPTION
**Pull request by FengZiYjun**
changes to preprocessor, trainer, inference & seq modeling
- [trainer] rename "batchify" to "make_batch" in trainer
- [trainer] pack (batch_x_pad, seq_len) into batch_x in make_batch for seq labeling, because seq length before pad is needed to make masks
- [trainer] unpack it in data_forward
- [model] shorten model definition
- [inference] build inference class. test_POS_pipeline.py is OK to infer
- [preprocessor] handle pickles in a nicer manner
- [FastNLP] add fastNLP.py as high-level API
- [tester][trainer] add cuda support
- [preprocess] fix label2index for padding label seq
- update README.md
- [test] add test_tester.py
- rename "action" to "core"
- [model] optimize cuda support in seq labeling model
- [test] add test data "pku" for chinese word seg
- test_tester.py and test_cws.py is OK to run!
- add train script of Chinese word segmentation and its config file
- [action] add k-means bucketing, partition sequences into buckets of nearly the same length
- [trainer] print train loss every 10 steps
- [loader] cws pku loader split sequence longer than max_seq_len into several shorter sequences

**ready to merge now** 

- 将Trainer中的batchify函数换名为make_batch
- POS Trainer的make_batch可以返回一个tuple，因为序列标注需要原始序列长度
- 上述tuple在data_forward函数里被unpack
- 模型的定义更短
- Inference类可以正常工作。test_POS_pipeline.py有infer函数，可以跑。
- 预处理器读写pickle的部分更简洁
- 包装了一个高层API，叫FastNLP类，仿照Spacy的API
- Trainer和Tester增加了cuda，但是需要优化
- fix preprocessor的 bug
- 更新了首页README.md的文件树
- 更多的测试样例
- action模块重命名为core
- 如果在模型的定义内部(如loss计算)需要创建新的变量，使用pytorch的`to`确保device一致
- 增加了中文分词pku数据集上的样例数据
- test_tester.py和test_cws.py可以正确运行
- 添加中文分词的训练脚本和相应的配置文件
- 添加bucket sampler， 每次采样出来的batch由长度相近的句子组成
- Trainer训练时每10个迭代打印一次loss
- 中文分词的data loader设定最大句子长度，将过长的句子切割为多个样例

准备合并，请做最后检查。

### CODE REVIEW

File: fastNLP/action/inference.py [hunk 1]
```diff
@@ -1,26 +1,116 @@
...
+
+    def predict(self, network, data):
+        """
+        Perform inference.
+        :param network:
+        :param data: multi-level lists of strings
+        :return result: the model outputs
+        """
+        # transform strings into indices
+        data = self.prepare_input(data)
```
Comment (FengZiYjun) [inline fastNLP/action/inference.py]: 传入的data是二级list，元素是string，这里将元素转为index

File: fastNLP/action/inference.py [hunk 2]
```diff
@@ -1,26 +1,116 @@
+import torch
+
+from fastNLP.action.action import Batchifier, SequentialSampler
+from fastNLP.loader.preprocess import load_pickle, DEFAULT_UNKNOWN_LABEL
+
+
 class Inference(object):
```
Comment (FengZiYjun) [inline fastNLP/action/inference.py]: Inference类用来使用模型做推断。关注点是输出结果。
（虽然有些操作与Trainer和Tester一致，这一点会在以后抽象出来。）

**Comment by FengZiYjun** [comment]
通过检查，合并。
